IliasDownloaderUniMA/__init__.py

```diff
@@ -73,6 +73,26 @@ def createIliasUrl(self, iliasid):
 		+ "&cmdNode=vi&baseClass=ilrepositorygui"
 
 
+	def extractIdFromUrl(self, url):
+		"""
+		Extracts the iliasid from a given ilias url.
+
+		:param      url:  ilias url containing a ref id
+		:type       url:  str
+
+		:returns:	iliasid of given course or object
+		:rtype:     int
+
+		:raises:	ValueError:  when an url without a iliasid is given.
+		"""
+
+		match = re.search(r"ref_id=(\d+)", url)
+		if match == None:
+			raise ValueError("No iliasid in given URL")
+
+		return int(match.group(1))
+
+
```

Comment:
Coded it like this so it'll be easier to test for me. But maybe I went a bit overboard with this one :sweat_smile: 
I think it'll mostly boil down to the question: Do you want short methods like createIliasUrl, extractIdFromUrl and addCourse to make methods easier to read, or do you want long all-in-one methods to cut down on code size.
It's your project, so I don't know what you'd prefer. Thought tbh because of createIliasUrl and scanCourses that you'd be more into the former design, maybe I parsed that wrong.

Reply:
Hm, my code design isn't really consistent and shouldn't serve as a rule tbh. It's still a quick-and-dirt solution in the most parts of the code. If someone is willed to write tests for all methods at a later point, I agree on your approach with short methods. But I still think an own method for this one and `createIliasUrl()` is overkill and both should be inlined into their respective places.